### PR TITLE
[FLINK-35149][cdc-composer] Fix DataSinkTranslator#sinkTo ignoring pre-write topology if not TwoPhaseCommittingSink

### PR DESCRIPTION
**Pull request by loserwang1024**
Current , when sink is not instanceof TwoPhaseCommittingSink, use input.transform rather than stream. It means that pre-write topology will be ignored.

```
private void sinkTo(
        DataStream<Event> input,
        Sink<Event> sink,
        String sinkName,
        OperatorID schemaOperatorID) {
    DataStream<Event> stream = input;
    // Pre write topology
    if (sink instanceof WithPreWriteTopology) {
        stream = ((WithPreWriteTopology<Event>) sink).addPreWriteTopology(stream);
    }

    if (sink instanceof TwoPhaseCommittingSink) {
        addCommittingTopology(sink, stream, sinkName, schemaOperatorID);
    } else {
        input.transform(
                SINK_WRITER_PREFIX + sinkName,
                CommittableMessageTypeInfo.noOutput(),
                new DataSinkWriterOperatorFactory<>(sink, schemaOperatorID));
    }
} 
```

（ps: the modify of StarRocksUtils just apply spotless)

### CODE REVIEW

**Comment by loserwang1024** [comment]
@PatrickRen , CC

**Comment by pvary** [comment]
Good catch @loserwang1024!

Could you please add a test case to prevent later code changes to revert this fix?

**Comment by loserwang1024** [comment]
> Could you please add a test case to prevent later code changes to revert this fix?

I'd like to, but it seems no pipeline sink which is WithPreWriteTopology but not TwoPhaseCommittingSink now unless i mock one.

**Comment by pvary** [comment]
> > Could you please add a test case to prevent later code changes to revert this fix?
> 
> I'd like to, but it seems no pipeline sink which is WithPreWriteTopology but not TwoPhaseCommittingSink now unless i mock one.

Could you define your own one in the test itself?
Then you have free hands what it does, and does not...

**Comment by loserwang1024** [comment]
> Could you define your own one in the test itself? Then you have free hands what it does, and does not...

Done it.

File: flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslatorTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+public class DataSinkTranslatorTest {
+
+    @Test
+    public void testPreWriteWithoutCommitSink() {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        ArrayList<Event> mockEvents = Lists.newArrayList(new EmptyEvent(), new EmptyEvent());
+        DataStreamSource<Event> inputStream = env.fromCollection(mockEvents);
+        DataSinkTranslator translator = new DataSinkTranslator();
+
+        String uid = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
```
[the remark this answers is not in the document]
Comment (loserwang1024) [reply flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslatorTest.java:48]: I've tried it before, but it shows that : 
java.lang.IllegalArgumentException: Node hash must be a 32 character String that describes a hex code.

File: flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslator.java [hunk 2]
```diff
@@ -71,7 +72,9 @@ public void translate(
         }
     }
 
-    private void sinkTo(
+    /** Only visible for test */
+    @VisibleForTesting
+    protected void sinkTo(
```
[the remark this answers is not in the document]
Comment (loserwang1024) [reply flink-cdc-composer/src/main/java/org/apache/flink/cdc/composer/flink/translator/DataSinkTranslator.java]: Thanks for your advice, it seems better.

**Comment by loserwang1024** [comment]
@PatrickRen , CC, Would you like to help me review this PR?

**Review by dingxin-tech — APPROVED** [review]

**Comment by yuxiqian** [comment]
Hi @loserwang1024, could you please backport this patch to `release-3.1` branch so that it could be released with CDC 3.1.1?

**Comment by loserwang1024** [comment]
> Hi @loserwang1024, could you please backport this patch to `release-3.1` branch so that it could be released with CDC 3.1.1?

I'd like to, but it haven't been merged to master.

**Review by PatrickRen — APPROVED** [review]
@loserwang1024 Thanks for the patch! LGTM

**Comment by PatrickRen** [comment]
@loserwang1024 Could you cherry-pick this commit to release-3.1 branch? Thanks

**Pull request merged**